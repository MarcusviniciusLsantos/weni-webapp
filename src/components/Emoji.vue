<!-- eslint-disable vue/multi-word-component-names -->
<template>
  <img
    :src="currentEmote"
    :title="name"
  />
</template>

<script>
export default {
  props: {
    name: {
      type: String,
    },
  },

  computed: {
    currentEmote() {
      const codes = {
        'Winking Face': 'u1F609',
        'Smiling Face with Smiling Eyes': 'u1F645',
      };

      const basePath = '../assets/emoji/';
      const imagePath = codes[this.name] ? `${basePath}${codes[this.name]}.png` : '';

      return imagePath;
    },
  },
};
</script>

<style lang="scss" scoped>
img {
  height: 1em;
  user-select: none;
  margin-bottom: -0.05em;
}
</style>
