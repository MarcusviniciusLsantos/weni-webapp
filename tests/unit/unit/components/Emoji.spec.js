import { shallowMount } from '@vue/test-utils';
import Emoji from '@/components/Emoji.vue';

describe('Emoji.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Emoji, {
      mocks: {
        $t: () => '',
      },
      props: {
        name: 'Winking Face',
      },
    });
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
