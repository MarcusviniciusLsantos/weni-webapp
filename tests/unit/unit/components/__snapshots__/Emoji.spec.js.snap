// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Emoji.vue > renders correctly 1`] = `
VueWrapper {
  "_emitted": {},
  "_emittedByOrder": [],
  "isFunctionalComponent": undefined,
}
`;

exports[`Emoji.vue renders correctly 1`] = `<img src="test-file-stub" title="Winking Face">`;
